File: kernel/pit.c
// based on information from here: https://wiki.osdev.org/Programmable_Interval_Timer

#include <kernel/io.h>
#include <kernel/pit.h>

#include <stdint.h>

void init_pit() {
    // mode 2
    outb(0x43, 0x36);

    // 100 Hz
    outb(0x40, 11931 & 0x00FF);
    outb(0x40, (11931 & 0xFF00) >> 8);
}

File: kernel/src/pit.zig
const io = @import("io.zig");

pub fn initialize() void {
    // mode 2
    io.outb(0x43, 0x36);

    // 100 Hz
    io.outb(0x40, 11931 & 0x00FF);
    io.outb(0x40, (11931 & 0xFF00) >> 8);
}
